metadata/src/test/java/org/apache/kafka/image/ProducerIdsImageTest.java

```diff
@@ -54,11 +54,24 @@ public class ProducerIdsImageTest {
             setBrokerId(3).
             setBrokerEpoch(100).
             setNextProducerId(789), (short) 0));
+        DELTA1_RECORDS.add(new ApiMessageAndVersion(new ProducerIdsRecord().
+            setBrokerId(3).
+            setBrokerEpoch(100).
+            setNextProducerId(780), (short) 0));
+        DELTA1_RECORDS.add(new ApiMessageAndVersion(new ProducerIdsRecord().
+            setBrokerId(3).
+            setBrokerEpoch(100).
+            setNextProducerId(785), (short) 0));
+        DELTA1_RECORDS.add(new ApiMessageAndVersion(new ProducerIdsRecord().
+            setBrokerId(2).
+            setBrokerEpoch(100).
+            setNextProducerId(460), (short) 0));
 
         DELTA1 = new ProducerIdsDelta(IMAGE1);
         RecordTestUtils.replayAll(DELTA1, DELTA1_RECORDS);
 
-        IMAGE2 = new ProducerIdsImage(789);
+        // producer ids must be increasing, but delta replay does not check this (only QuorumController replay)
```

Comment:
@rondagostino what are your thoughts on this check then? should I remove it as well?

Reply:
I guess we need to decide what to do with these types of cases.  The overall issue is that right now the various `*Delta` classes will replay pretty much whatever you give them without complaining -- including stuff that the actual `QuorumController` would never send and that would be unable to replay.

We could keep these changes as opposed to just silently deleting them from this PR/ignoring the issue that they point to.  If we did that then at a minimum we should add copious comments about what the test is doing and why we expect it to never happen/why we are testing it anyway.

But it feels to me that these are strange cases to test, and it might just be best to open a JIRA ticket to make these various `*Delta` classes actually reject illegal stuff like this so that we get an exception closer to the point where the issue actually originates.

Reply:
let's go with the latter then, we'll need to iron out validation for the sake of the metadata recovery tool anyways